bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/discovery/InboxResource.java

```diff
@@ -100,7 +103,9 @@ public Response approve(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(valu
         try {
             thing = inbox.approve(thingUIDObject, notEmptyLabel);
         } catch (IllegalArgumentException e) {
-            return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Thing not found in inbox");
+            logger.error("Thing {} unable to be approved: {}", thingUID, e.getLocalizedMessage());
+            return JSONResponse.createErrorResponse(Status.NOT_FOUND,
+                    "Thing unable to be approved: " + e.getLocalizedMessage());
```

Comment:
I agree with the log message but adding the message of an exception to the REST response seems wrong to me. Exceptions and its messages are part of the internals and should not be exposed to the caller of a REST API.
Or does the inbox approve methods state that the exception message (if IAE) is exposed to external callers and this message must not contain any internal information?